src/pages/experiments/[experimentId]/plots-and-tables/marker-heatmap/index.jsx

```diff
@@ -0,0 +1,307 @@
+import Header from '../../../../../components/plots/Header';
+import { generateSpec } from '../../../../../utils/plotSpecs/generateHeatmapSpec';
+import { loadGeneExpression, loadMarkerGenes } from '../../../../../redux/actions/genes';
+import { loadCellSets } from '../../../../../redux/actions/cellSets';
+import PlatformError from '../../../../../components/PlatformError';
+import Loader from '../../../../../components/Loader';
+import populateHeatmapData from '../../../../../components/plots/helpers/populateHeatmapData';
+import HeatmapControls from '../../../../../components/plots/styling/heatmap/HeatmapControls';
+
+const { Text } = Typography;
+const { Panel } = Collapse;
+
+const plotUuid = 'markerHeatmapPlotMain';
+const plotType = 'markerHeatmap';
+
+const route = {
+  path: 'marker-heatmap',
+  breadcrumbName: 'Marker-Heatmap',
+};
+const MarkerHeatmap = ({ experimentId }) => {
+  const dispatch = useDispatch();
+  const config = useSelector((state) => state.componentConfig[plotUuid]?.config);
+
+  const { expression: expressionData } = useSelector((state) => state.genes);
+  const { error, loading } = expressionData;
+  const cellSets = useSelector((state) => state.cellSets);
+  const { hierarchy, properties } = cellSets;
+  const selectedGenes = useSelector((state) => state.genes.expression.views[plotUuid]?.data) || [];
+  const [currentSearchedGenes, setCurrentSearchedGenes] = useState([]);
+  const {
+    loading: loadingMarkerGenes,
+    error: errorMarkerGenes,
+  } = useSelector((state) => state.genes.markers);
+  const [vegaSpec, setVegaSpec] = useState();
+  const louvainClustersResolution = useSelector(
+    (state) => state.experimentSettings.processing
+      .configureEmbedding?.clusteringSettings.methodSettings.louvain.resolution,
+  ) || false;
+
+  useEffect(() => {
+    if (!louvainClustersResolution) {
+      dispatch(loadProcessingSettings(experimentId));
+    }
+    dispatch(loadPlotConfig(experimentId, plotUuid, plotType));
+    if (!cellSets.hierarchy.length) {
+      dispatch(loadCellSets(experimentId));
+    }
+  }, []);
+
+  useEffect(() => {
+    if (louvainClustersResolution && config) {
+      dispatch(loadMarkerGenes(experimentId, louvainClustersResolution, plotUuid, config.numGenes));
+    }
+  }, [louvainClustersResolution, hierarchy, config?.numGenes]);
+  const sortGenes = (newGenes) => {
+    const clusters = hierarchy.find((cluster) => cluster.key === config.selectedCellSet).children;
+
+    const getCellIdsForCluster = (clusterId) => properties[clusterId].cellIds;
+
+    const getAverageExpressionForGene = (gene, currentCellIds) => {
+      const expressionValues = expressionData.data[gene].rawExpression.expression;
+      let totalValue = 0;
+      currentCellIds.forEach((cellId) => {
+        totalValue += expressionValues[cellId];
+      });
+      return totalValue / currentCellIds.size;
+    };
+
+    const getClusterForGene = (gene) => {
+      let maxAverageExpression = [0];
```

Comment:
I think that the default value of `maxAverageExpression` as a list with 0 is confusing -- what you technically have is a list with 2 items, where the first index shows the value of the max expression and the second shows the index of the cluster where this value occurs. 
The code will be cleaner if you either:
- use 2 variables for this: `maxAvgExpression`, just a number denoting max expression; and `maxExpressionClusterIndex` denoting the cluster index when it occured. Then in line 90 you just return `maxExpressionClusterIndex`.
- use 1 object for this, called `maxAvgExpression` that looks like something like this:
```
let maxAvgExpression = {"expression": 0, "clusterIndex": -1}
```
then on line 90 you return maxAvgExpression["clusterIndex"]

Reply:
another thing: are you sure that it is not possible to have negative expression? If negative expression is possible, then setting default maxExpression to 0 is incorrect.